aerogear/mobile-security-service
# Created a custom HTTP error package

Created custom package to return HTTP errors in a concise manner

## Motivation

JIRA: https://issues.jboss.org/browse/AEROGEAR-839

## What

Created a custom package to wrap Echo's HTTP error handler and return HTTP errors with a JSON body.

## Why

The default HTTP error handling that comes with Echo is a bit verbose and could do with improvements.

Example:

```go
return echo.NewHTTPError(http.StatusUnauthorized, "Please provide valid credentials")
```

This could be improved to be like this:

```go
return httperrors.Unauthorized(c, "Please provide valid credentials")
```

## How

1. Created a custom package in `pkg/httperrors/httperrors.go`.
2. Created functions for the common HTTP status codes (e.g. 404 Not Found)

## Verification Steps
Add the steps required to check this change. Following an example.
 
1. Go to `cmd/mobile-security-service-server/main.go`
2. Import the `httperrors` package:

```go
import (
	"github.com/aerogear/mobile-security-service-server/pkg/httperrors"
        ...
)
```

3. Add the following code block in `main()` after `e := echo.New()`:

```go
e.GET("/test", func(c echo.Context) error {
        return httperrors.NotFound(c, "Test")
})
```

4. Run `go run cmd/mobile-security-service-server/main.go` to start the server.

5. Go to [http://localhost:3000/test](http://localhost:3000/test) (replace with your own port). You should see a response body like this:

```json
{
    "message": "Test",
    "statusCode": 404
}
```

6. You can repeat this with the other status codes.

## Checklist:


## Progress
